core/src/main/java/org/apache/iceberg/DataFilesTable.java

```diff
@@ -109,11 +112,20 @@ public long targetSplitSize() {
       Expression filter = ignoreResiduals ? Expressions.alwaysTrue() : rowFilter;
       ResidualEvaluator residuals = ResidualEvaluator.unpartitioned(filter);
 
+      // use an inclusive projection to remove the partition name prefix and filter out any non-partition expressions
+      Expression partitionFilter = Projections
+          .inclusive(transformSpec(fileSchema, table().spec()), caseSensitive)
+          .project(rowFilter);
+
+      ManifestEvaluator manifestEval = ManifestEvaluator.forPartitionFilter(partitionFilter, table().spec(),
+          caseSensitive);
```

Comment:
Nit: I'd probably wrap so that all the arguments are on the same line. That seems more readable to me.

Reply:
You are right, done